src/Performance/Minibench.purs

```diff
@@ -88,4 +107,55 @@ bench
   :: forall eff a
    . (Unit -> a)
   -> Eff (console :: CONSOLE | eff) Unit
-bench = benchWith 1000
+bench = benchWith defaultNumSamples
+
+-- | Run an asynchronous function n times sequentially, calling the given
+-- | callback with the array of runtimes after all runs are completed.
+runAsync
+  :: forall e a
+   . Int
+  -> (Eff (console :: CONSOLE | e) Unit -> Eff (console :: CONSOLE | e) Unit)
+  -> (Array Number -> Eff (console :: CONSOLE | e) Unit)
+  -> Eff (console :: CONSOLE | e) Unit
+runAsync n f done = runAsync' n []
+  where
+    runAsync' 0 runs = done runs
+    runAsync' m runs = do
+      t1 <- runEffFn1 hrTime [0, 0]
+      f do
+        t2 <- runEffFn1 hrTime t1
+        void $ runAsync' (m - 1) ((fromHrTime t2):runs)
+
+-- | Estimate the running time of an asynchronous function and print a summary
+-- | to the console, specifying the number of samples to take. More samples will
+-- | give a better estimate of both mean and standard deviation, but will
+-- | increase running time.
+benchAsyncWith
+  :: forall e a
+   . Int
+  -> (Eff (console :: CONSOLE | e) Unit -> Eff (console :: CONSOLE | e) Unit)
+  -> Eff (console :: CONSOLE | e) Unit
+  -> Eff (console :: CONSOLE | e) Unit
+benchAsyncWith n f done =
+  runAsync n f \runs -> logShow (results runs) >>= \_ -> done
+
+-- | Estimate the running time of an asynchronous function and print a summary
+-- | to the console, by running the function 1000 times.
+-- |
+-- | For example:
+-- |
+-- | ```
+-- | > import Data.Array
+-- | > import Data.Foldable
+-- | > import Performance.Minibench
+-- | > benchAsync \done -> sum (1 .. 10000) >>= \_ -> done
```

Comment:
This example doesn't seem to type-check.

Reply:
heh, oops, sorry about that.  Will update.